pandas/_libs/lib.pyx

```diff
@@ -2359,3 +2356,103 @@ def fast_multiget(dict mapping, ndarray keys, default=np.nan):
             output[i] = default
 
     return maybe_convert_objects(output)
+
+
+@cython.wraparound(False)
+@cython.boundscheck(False)
+cdef inline void convert_and_set_item(object item, Py_ssize_t index,
+                                      object[:] result,
+                                      bint keep_trivial_numbers):
+    cdef:
+        bint do_convert = 1
+        double double_item
+
+    if keep_trivial_numbers:
+        if isinstance(item, int) and Py_SIZE(item) < 2:
```

Comment:
What does the pysize check correspond to in python-land?

Reply:
There's no direct Python equivalent for `Py_SIZE`.

It's actually a shortcut for speeding up the check if `item` (as integer) is zero. In Python 3, `int`-s are internally able to do long math, but if they're having several `int32`-sized "digits" they cannot be zero, so converting them to a `long long` or anything else just to check for zero adds extra overhead.

So this is some CPython implementation details being used for a slight speed increase of simple `if isinstance(item, int) and item == 0` check.

Reply:
Got it, thanks.  Please comment so this is clear to the next reader.

Would be liable to break on PyPy or some other implementation?

Reply:
I'm not sure how well this Cython code could be compiled for PyPy (given the amount of stuff Pandas already `cimport` from `cpython`), but potentially yes, this is implementation details which might break. But quick search says that Cython backend for PyPy is still not there, are [their wiki](https://github.com/cython/cython/wiki/enhancements-pypy) being inaccurate?

Will add a comment tomorrow.

Reply:
@jreback do we have a policy on relying on CPython implementation details?  It seems like the sort of thing we'd avoid.

Reply:
we don't really care much for PyPy , though we do take patches.

Reply:
We've dropped usage of `Py_SIZE` (no more implementation specific here).